starknet-rpc-test/src/lib.rs

```diff
@@ -1,17 +1,43 @@
+#![feature(assert_matches)]
+
 use std::cell::Cell;
 use std::fmt::Debug;
+use std::net::TcpListener;
 use std::path::Path;
 use std::process::{Child, Command, Stdio};
 
 use anyhow::anyhow;
+use constants::{MAX_PORT, MIN_PORT};
 use derive_more::Display;
+use lazy_static::lazy_static;
 use reqwest::header::CONTENT_TYPE;
 use reqwest::{Client, Response};
 use serde_json::json;
+use starknet_accounts::{Execution, SingleOwnerAccount};
+use starknet_core::types::InvokeTransactionResult;
 use starknet_providers::jsonrpc::{HttpTransport, JsonRpcClient};
 use starknet_providers::Provider;
+use starknet_signers::LocalWallet;
+use thiserror::Error;
+use tokio::sync::Mutex;
 use url::Url;
 
+/// Constants (addresses, contracts...)
+pub mod constants;
+/// Starknet related utilities
+pub mod utils;
+
+type TransactionExecution<'a> = Execution<'a, SingleOwnerAccount<&'a JsonRpcClient<HttpTransport>, LocalWallet>>;
+
+lazy_static! {
+        /// This is to prevent TOCTOU errors; i.e. one background madara node might find one
+        /// port to be free, and while it's trying to start listening to it, another instance
+        /// finds that it's free and tries occupying it
+        /// Using the mutex in `get_free_port_listener` might be safer than using no mutex at all,
+        /// but not sufficiently safe
+        static ref BACKGROUND_MADARA_MUTEX: Mutex<()> = Mutex::new(());
```

Comment:
`Mutex::new(())` is `const` Is it possible to make it a const rather than a lazy_static, or does it cause trouble with the way we are using it?